File: Codeforces/Codeforces Round 894 (Div. 3)/D_Ice_Cream_Balls.cpp
#include<iostream>

using namespace std;
using ull = unsigned long long;

void solve() {
    ull input;
    cin >> input;
    ull l = 2, r = 1414213563;
    while (l < r) {
        ull mid = l + (r - l) / 2;
        if (mid * (mid - 1) / 2 <= input) l = mid + 1;
        else r = mid;
    }
    l--;
    cout << input - (l - 1) * l / 2 + l << '\n';
}

int main() {
    int t;
    cin >> t;
    while (t--) solve();
    return 0;
}

File: Codeforces/Codeforces Round 894 (Div. 3)/D_Ice_Cream_Balls.py
for _ in range(int(input())):
    a = int(input())
    l = 2
    r = 1414213563
    while l < r:
        mid = l + (r - l) // 2
        if mid * (mid - 1) <= a * 2:
            l = mid + 1
        else:
            r = mid
    l -= 1
    print(a - (l - 1) * l // 2 + l)
